lib/grandpa/message_handler.go

```diff
@@ -78,7 +77,31 @@ func (h *MessageHandler) handleMessage(from peer.ID, m GrandpaMessage) (network.
 	}
 }
 
+func (h *MessageHandler) notifyNeighbors(neighbourMessage *NeighbourPacketV1) {
```

Comment:
I think this should return an error.  and the calling function can just decide to `log.Warnf` any error that comes back.